File: TypeScript/Algorithm/Algorithm I/283. Move Zeroes.ts
function moveZeroes(nums: number[]): void {
    const numsLength: number = nums.length;
    let nonZeroIndex: number = 0;

    for (let index: number = 0; index < numsLength; index++) {
        if (nums[index]) {
            if (index !== nonZeroIndex) {
                nums[nonZeroIndex] = nums[index];
                nums[index] = 0;
            }
            nonZeroIndex++;
        }
    }
}

File: Java/Algorithm/Algorithm I/283. Move Zeroes.java
class Solution {

  /**
   * Moves all zeroes to the end of an integer array while maintaining the relative order of non-zero elements.
   *
   * @param nums An array of integers containing zeroes and non-zeroes.
   */
  public void moveZeroes(int[] nums) {
    int NUMS_LENGTH = nums.length;
    int nonZeroIndex = 0;

    for (int index = 0; index < NUMS_LENGTH; index++) {
      if (nums[index] != 0) {
        int tempValue = nums[index];

        nums[index] = nums[nonZeroIndex];
        nums[nonZeroIndex] = tempValue;

        nonZeroIndex++;
      }
    }
  }
}
